drivers/periph_common/init.c

```diff
@@ -49,21 +49,24 @@
 #ifdef MODULE_PERIPH_INIT_PTP
 #include "periph/ptp.h"
 #endif
+#ifdef MODULE_PERIPH_INIT_PIO
+#include "periph/pio.h"
+#endif
 #endif /* MODULE_PERIPH_INIT */
 
 void periph_init(void)
 {
 #ifdef MODULE_PERIPH_INIT
     /* initialize configured I2C devices */
 #ifdef MODULE_PERIPH_INIT_I2C
-    for (unsigned i = 0; i < I2C_NUMOF; i++) {
+    for (int i = 0; i < (int)I2C_NUMOF; i++) {
```

Comment:
If `I2C_NUMOF` is 0, the compiler warns me that I maybe did something I did not intend to do.
```
RIOT/drivers/periph_common/init.c:62:28: error: comparison of unsigned expression in '< 0' is always false [-Werror=type-limits]
   62 |     for (unsigned i = 0; i < (unsigned)I2C_NUMOF; i++) {
      | 
```

Reply:
Wouldn't some guard like 
```suggestion
    #if I2C_NUMOF > 0
    for (unsigned i = 0; i < I2C_NUMOF; i++) {
```
or 
```suggestion
    if( I2C_NUMOF > 0 ) for (unsigned i = 0; i < I2C_NUMOF; i++) {
```
help 
or notusing of
"MODULE_PERIPH_INIT_I2C"

Reply:
I tried both of your suggestions
First suggestion unfortunately does not work because `I2C_NUMOF` expands to `sizeof` which is not understood by the preprocessor. The second suggestions throws the same compiler warning as before.

> "MODULE_PERIPH_INIT_I2C"
> but this might need more modifications to the build system

Do you have something on your mind? I don´t know if this would be worth.
I think the change is reasonable, if I make it a separate commit.

Reply:
> First suggestion unfortunately does not work because I2C_NUMOF

Sorry, `PIO_NUMOF` expands to `sizeof`. It probably does work for `I2C_NUMOF` and `SPI_NUMOF`

Reply:
something like 
```
char t[0]; 

static unsigned _ident(unsigned b){return b;};

void test2(void){
    unsigned i; 
    if(i < _ident( sizeof(t) )){
        (void) 0;
    }
}
```
makes the compiler not warn (but still optimize everything away) without using compiler specifics

Reply:
nice hack!